File: 283.move_zeros.ts
function moveZeroes(nums: number[]): void {
    let i: number = 0;
    let j: number = i;
    while (i < nums.length) {
        if (nums[i] == 0) {
            j = i
            while (j < nums.length) {
                if (nums[j] != 0) {
                    nums[i] = nums[j];
                    nums[j] = 0;
                    break;
                }
                j++;
            }
        }
        i++;
    }
    console.log(nums)
};

moveZeroes([0, 1, 0, 3, 12])

File: 283.move_zeros.go
package main

func moveZeroes(nums []int) {
	i := 0
	j := 0
	for i < len(nums) {
		if nums[i] == 0 {
			j = i
			for j < len(nums) {
				if nums[j] == 0 {
					j++
				} else {
					nums[i] = nums[j]
					nums[j] = 0
					break
				}
			}
		}
		i++
	}
}
